climada/hazard/tc_tracks.py

```diff
@@ -1851,3 +1854,64 @@ def set_category(max_sus_wind, wind_unit='kn', saffir_scale=None):
         return (np.argwhere(max_wind < saffir_scale) - 1)[0][0]
     except IndexError:
         return -1
+
+def _make_line_from_track(track, split_lines_antimeridian=True):
+    """Auxiliary function to convert a track into a Point (len==1), LineString
+    (standard case), or MultiLineString (crossing antimeridian).
+
+    Parameters
+    ----------
+    track : xr.Dataset
+        A track as present as an element of TCTracks.data
+    
+    Returns
+    -------
+    line : Point, LineString, MultiLineString
+        A shapely feature representing a single TC track.
+    """
+    lon = u_coord.lon_normalize(track.lon.values.copy(), center=0.0)
+    lat = track.lat.values
+    # LineString only works with more than one lat/lon pair
+    if lon.size == 1:
+        return Point(lon, lat)
+    extent = u_coord.lon_bounds(lon)
+    if split_lines_antimeridian and extent[0] <= 180 <= extent[1]:
+        # in this case split into multilines
+        hem_west = lon < 0
+        # if all points on one side are +180 or -180, keep as a single line
+        if np.all(lon[hem_west] == -180) or np.all(lon[~hem_west] == 180):
+            if np.all(lon[hem_west] == -180):
+                lon[hem_west] = 180
+            else:
+                lon[~hem_west] = -180
+            return LineString(np.c_[lon, lat])
+        # otherwise, split lines
+        split_idx = np.concatenate((np.array([0]),
+                                    np.where(np.diff(lon > 0))[0] + 1,
```

Comment:
Both cases will not come to pass, since there is a check on the `extent` before this. Not even the (enormously odd) case of all values lying on the antimeridian would be met, since that is caught beforehand.